src/Components/PostsComponent/DisplayPosts.jsx

```diff
@@ -14,42 +56,55 @@ class DisplayPosts extends React.Component {
     this.setState({ loading: false })
   }
 
-  // async componentDidUpdate(prevState) {
-  //   const postsFromDb = await getAllPosts()
-  //   if (prevState !== postsFromDb) {
-  //     this.setState({ posts_arr: postsFromDb })
-  //   }
-  // }
-
   render() {
     const posts = [...this.state.posts_arr]
 
-    let i = 0
+    // let i = 0
 
     if (this.state.loading) {
       return "loading..."
     }
 
     return (
       <div>
-        {this.state.posts_arr && this.state.posts_arr.length ? (
-          <div>
-            {posts.map((post) => {
-              return (
-                <div key={i++}>
-                  <h1>{post.title}</h1>
-                  <h2>{post.desc}</h2>
-                  <h3>
-                    posted at{" "}
-                    {new Date(Number(post.createdAt)).toLocaleTimeString()}.
-                  </h3>
-                </div>
-              )
-            })}
-          </div>
-        ) : (
-          "No post to display"
-        )}
+        <CreatePost
+          title={this.state.titleValue}
+          desc={this.state.descValue}
+          titleInpHandler={this.titleInputHandler}
```

Comment:
It's better to have input states inside that respective component. Do you have any advantage of lifting the input state higher here?

Reply:
Done sir.